packages/web3-eth-personal/test/integration/personal.test.ts

```diff
@@ -0,0 +1,125 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+import { isHexStrict } from 'web3-validator';
+import { toChecksumAddress } from 'web3-utils';
+import { EthPersonal } from '../../src/index';
+import { importedAccount } from '../config/personal.test.config';
+import {
+	getSystemTestBackend,
+	getSystemTestAccounts,
+	getSystemTestProvider,
+	itIf,
+} from '../fixtures/system_test_utils';
+
+describe('personal integration tests', () => {
+	let ethPersonal: EthPersonal;
+	let accounts: string[];
+	beforeAll(() => {
+		const clientUrl = getSystemTestProvider();
+		ethPersonal = new EthPersonal(clientUrl);
+	});
+	beforeEach(async () => {
+		accounts = await getSystemTestAccounts();
+	});
+	it('new account', async () => {
+		const newAccount = await ethPersonal.newAccount('!@superpassword');
+		expect(isHexStrict(newAccount)).toBe(true);
+	});
+
+	itIf(getSystemTestBackend() === 'geth')('ecRecover', async () => {
+		// ganache does not support ecRecover
+		const signature = await ethPersonal.sign('0x2313', accounts[0], '');
+		const publicKey = await ethPersonal.ecRecover('0x2313', signature); // ecRecover is returning all lowercase
+		// eslint-disable-next-line jest/no-standalone-expect
+		expect(toChecksumAddress(publicKey)).toBe(toChecksumAddress(accounts[0]));
+	});
+
+	it('lock account', async () => {
+		const lockAccount = await ethPersonal.lockAccount(accounts[1]);
+		expect(lockAccount).toBe(true);
+	});
+
+	it('unlock account', async () => {
+		const key = accounts[0];
+		const unlockedAccount = await ethPersonal.unlockAccount(key, '', 100000);
+		expect(unlockedAccount).toBe(true);
+	});
```

Comment:
Same here. These tests files will run as `integration` and `e2e`, so the exact behavior of the function should be tested. If not directly in a test, may be through some other tests. 

e.g. In contract deployment tests I didn't tested if the contract was actually deployed or not, because it was indirectly tested in the methods and events case. But the behavior of contract deployment was fully tested end-to-end.